python_modules/dagster/dagster/components/core/tree.py

```diff
@@ -0,0 +1,80 @@
+import importlib
+from functools import cached_property
+from pathlib import Path
+from types import ModuleType
+from typing import Optional
+
+from dagster_shared.record import record
+from typing_extensions import TypeVar
+
+from dagster._core.definitions.definitions_class import Definitions
+from dagster.components.component.component import Component
+from dagster.components.core.context import ComponentLoadContext, use_component_load_context
+from dagster.components.core.defs_module import DefsFolderComponent
+from dagster.components.core.load_defs import get_library_json_enriched_defs
+
+PLUGIN_COMPONENT_TYPES_JSON_METADATA_KEY = "plugin_component_types_json"
+
+TComponent = TypeVar("TComponent", bound=Component)
+
+
+@record(
+    checked=False,  # cant handle ModuleType
+)
+class ComponentTree:
+    defs_module: ModuleType
+    project_root: Path
+
+    @staticmethod
+    def find(path: Optional[Path] = None):
```

Comment:
`find` is super generic for what this is doing and I think obscurees.

This means "I will take a path that is in the context of a dagster project and give you the root component tree" correct?

`def load(path_within_project: Path)`

Would be more clear to me

I actually think defaulting to `Path.cwd` might cause more confusion. Not a big lift to force users to call `Path.cwd` and they will be able to understand what is happening.